usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -494,8 +523,20 @@
                 UPPER(bs.sub_place_of_perform_country_co) <> 'USA'
             ) THEN NULL
             ELSE COALESCE(pop_cd_state_grouped.congressional_district_no, pop_zips.congressional_district_no, pop_cd_zips_grouped.congressional_district_no, pop_cd_city_grouped.congressional_district_no, pop_cd_county_grouped.congressional_district_no)
-        END) AS sub_place_of_performance_congressional_current
-
+        END) AS sub_place_of_performance_congressional_current,
+        rl_state_fips.fips as legal_entity_state_fips,
+        pop_state_fips.fips as place_of_perform_state_fips,
+        case
+            when rl_state_fips.fips is null or rl_county_fips.county_numeric is null
+                then NULL
+            else concat(rl_state_fips.fips, rl_county_fips.county_numeric)
+        end as legal_entity_county_fips,
```

Comment:
These case statements are unnecessary. In SparkSQL, if any of the inputs to a CONCAT function are NULL, the output will be null.

Reply:
Ah good catch! Thank you.